packages/image_picker/image_picker_linux/lib/image_picker_linux.dart

```diff
@@ -0,0 +1,151 @@
+class ImagePickerLinux extends CameraDelegatingImagePickerPlatform {
+  /// Constructs a platform implementation.
+  ImagePickerLinux();
+
+  /// The file selector used to prompt the user to select images or videos.
+  @visibleForTesting
+  static FileSelectorPlatform fileSelector = FileSelectorLinux();
+
+  /// Registers this class as the default instance of [ImagePickerPlatform].
+  static void registerWith() {
+    ImagePickerPlatform.instance = ImagePickerLinux();
+  }
+
+  @override
+  Future<PickedFile?> pickImage({
+    required ImageSource source,
+    double? maxWidth,
+    double? maxHeight,
+    int? imageQuality,
+    CameraDevice preferredCameraDevice = CameraDevice.rear,
+  }) async {
+    final XFile? file = await getImageFromSource(
+        source: source,
+        options: ImagePickerOptions(
+            maxWidth: maxWidth,
+            maxHeight: maxHeight,
+            imageQuality: imageQuality,
+            preferredCameraDevice: preferredCameraDevice));
+    if (file != null) {
+      return PickedFile(file.path);
+    }
+    return null;
+  }
+
+  @override
+  Future<PickedFile?> pickVideo({
+    required ImageSource source,
+    CameraDevice preferredCameraDevice = CameraDevice.rear,
+    Duration? maxDuration,
+  }) async {
+    final XFile? file = await getVideo(
+        source: source,
+        preferredCameraDevice: preferredCameraDevice,
+        maxDuration: maxDuration);
+    if (file != null) {
+      return PickedFile(file.path);
+    }
+    return null;
+  }
+
+  @override
+  Future<XFile?> getImage({
+    required ImageSource source,
+    double? maxWidth,
+    double? maxHeight,
+    int? imageQuality,
+    CameraDevice preferredCameraDevice = CameraDevice.rear,
+  }) async {
+    return getImageFromSource(
+        source: source,
+        options: ImagePickerOptions(
+            maxWidth: maxWidth,
+            maxHeight: maxHeight,
+            imageQuality: imageQuality,
+            preferredCameraDevice: preferredCameraDevice));
+  }
+
+  // [ImagePickerOptions] options are not currently supported. If any
+  // of its fields are set, they will be silently ignored.
+  //
+  // If source is `ImageSource.camera`, a `StateError` will be thrown
+  // unless a [cameraDelegate] is set.
+  @override
+  Future<XFile?> getImageFromSource({
+    required ImageSource source,
+    ImagePickerOptions options = const ImagePickerOptions(),
+  }) async {
+    switch (source) {
+      case ImageSource.camera:
+        return super.getImageFromSource(source: source);
+      case ImageSource.gallery:
+        const XTypeGroup typeGroup =
+            XTypeGroup(label: 'Images', mimeTypes: <String>['image/*']);
+        final XFile? file = await fileSelector
+            .openFile(acceptedTypeGroups: <XTypeGroup>[typeGroup]);
+        return file;
+    }
+    // Ensure that there's a fallback in case a new source is added.
+    // ignore: dead_code
+    throw UnimplementedError('Unknown ImageSource: $source');
```

Comment:
Not that it's a problem, but in what case is this likely to be an issue? Our lints should catch this (as I'm assuming they do based on the ignore), and I *assume* ImageSource is defined in the same package?

Reply:
It's not in the same package actually, it's from the platform interface. Our lints would catch it in CI, and we'd update this, but since they are separate packages and we don't consider adding enum values to be a breaking change in most cases, it's possible for someone to run a version `image_picker_linux` that hasn't been updated with a platform interface and app-facing package that has.

Reply:
Ah that's right - makes sense.